pandas/tests/resample/test_base.py

```diff
@@ -277,15 +277,20 @@ def test_resample_size_empty_dataframe(freq, empty_frame_dti):
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.filterwarnings(r"ignore:PeriodDtype\[B\] is deprecated:FutureWarning")
 @pytest.mark.parametrize("index", tm.all_timeseries_index_generator(0))
 @pytest.mark.parametrize("dtype", [float, int, object, "datetime64[ns]"])
 def test_resample_empty_dtypes(index, dtype, resample_method):
     # Empty series were sometimes causing a segfault (for the functions
     # with Cython bounds-checking disabled) or an IndexError.  We just run
     # them to ensure they no longer do.  (GH #10228)
+    if isinstance(index, PeriodIndex):
+        # GH#53511
+        index = PeriodIndex([], freq="B", name=index.name)
```

Comment:
why is this necessary? it looks like, if `index` is an empty `PeriodIndex` with `freq` 'B'`, then you're re-creating it?

Reply:
per matt's request i changed the fixture, so now index.freq=="D"